zebra-chain/src/sapling/commitment.rs

```diff
@@ -199,14 +261,22 @@ impl ValueCommitment {
     /// Generate a new _ValueCommitment_.
     ///
     /// https://zips.z.cash/protocol/protocol.pdf#concretehomomorphiccommit
-    #[allow(non_snake_case)]
-    pub fn new<T>(csprng: &mut T, value: Amount<NonNegative>) -> Self
+    pub fn randomized<T>(csprng: &mut T, value: Amount) -> Self
     where
         T: RngCore + CryptoRng,
     {
-        let v = jubjub::Fr::from(value);
         let rcv = generate_trapdoor(csprng);
 
+        Self::new(rcv, value)
+    }
+
+    /// Generate a new _ValueCommitment_ from an existing _rcv_ on a _value_.
+    ///
+    /// https://zips.z.cash/protocol/protocol.pdf#concretehomomorphiccommit
+    #[allow(non_snake_case)]
+    pub fn new(rcv: jubjub::Fr, value: Amount) -> Self {
+        let v = jubjub::Fr::from(value);
+
         // TODO: These generator points can be generated once somewhere else to
         // avoid having to recompute them on every new commitment.
         let V = find_group_hash(*b"Zcash_cv", b"v");
```

Comment:
Should we make these into a lazy_static?

Reply:
Unfortunately `lazy_static` doesn't quite do what we want:

![image](https://user-images.githubusercontent.com/552961/95156229-0a8bf300-0764-11eb-9c07-ef2d42665f9a.png)

Reply:
Ah, so there's a trick here… you have to Deref the struct created by the lazy_static macro. (Confusingly, the struct has the same name as the variable it replaces, but is its own unique type.)

Sometimes, taking the struct as a reference using `&V` works. But it seems like `V.as_ref()` is more reliable:
https://github.com/ZcashFoundation/zebra/pull/1116/files#diff-bc19effbc9086e335d897caab9487b76R58

Reply:
> But it seems like V.as_ref() is more reliable:

This is because it is using the method resolution rules which are closely interwined with `Deref` in rust. Which we could leverage this directly by using `V.mul(v)`